src/hypothesis/extra/numpy.py

```diff
@@ -73,20 +105,223 @@ def is_scalar(spec):
+            sizes = sorted(set(s // 8 for s in sizes))
+    strat = st.sampled_from(sizes)
+    if '{}' not in kind:
+        kind += '{}'
+    if endianness == '?':
+        return strat.map(('<' + kind).format) | strat.map(('>' + kind).format)
+    return strat.map((endianness + kind).format)
+
+
+@defines_dtype_strategy
+def unsigned_integer_dtypes(endianness='?', sizes=(8, 16, 32, 64)):
+    """Return a strategy for unsigned integer dtypes.
+
+    endianness may be ``<`` for little-endian, ``>`` for big-endian,
+    ``=`` for native byte order, or ``?`` to allow either byte order.
+    This argument only applies to dtypes of more than one byte.
+
+    sizes must be a collection of integer sizes in bits.  The default
+    (8, 16, 32, 64) covers the full range of sizes.
+
+    """
+    return dtype_factory('u', sizes, (8, 16, 32, 64), endianness)
+
+
+@defines_dtype_strategy
+def integer_dtypes(endianness='?', sizes=(8, 16, 32, 64)):
+    """Return a strategy for signed integer dtypes.
+
+    endianness and sizes are treated as for `unsigned_integer_dtypes`.
+
+    """
+    return dtype_factory('i', sizes, (8, 16, 32, 64), endianness)
+
+
+@defines_dtype_strategy
+def floating_dtypes(endianness='?', sizes=(16, 32, 64)):
+    """Return a strategy for floating-point dtypes.
+
+    sizes is the size in bits of floating-point number.  Some machines support
+    96- or 128-bit floats, but these are not generated by default.
+
+    Larger floats (96 and 128 bit real parts) are not supported on all
+    platforms and therefore disabled by default.  To generate these dtypes,
+    include these values in the sizes argument.
+
+    """
+    return dtype_factory('f', sizes, (16, 32, 64, 96, 128), endianness)
+
+
+@defines_dtype_strategy
+def complex_number_dtypes(endianness='?', sizes=(64, 128)):
+    """Return a strategy complex-number dtypes.
+
+    sizes is the total size in bits of a complex number, which consists
+    of two floats.  Complex halfs (a 16-bit real part) are not supported
+    by numpy and will not be generated by this strategy.
+
+    """
+    return dtype_factory('c', sizes, (64, 128, 192, 256), endianness)
+
+
+@defines_dtype_strategy
+def datetime64_dtypes(max_period='Y', min_period='ns', endianness='?'):
+    """Return a strategy for datetime64 dtypes, with various precisions from
+    year to attosecond."""
+    tr = TIME_RESOLUTIONS
+    res = tr[tr.index(max_period):tr.index(min_period) + 1]
+    return dtype_factory('datetime64[{}]', res, TIME_RESOLUTIONS, endianness)
+
+
+@defines_dtype_strategy
+def timedelta64_dtypes(max_period='Y', min_period='ns', endianness='?'):
+    """Return a strategy for timedelta64 dtypes, with various precisions from
+    year to attosecond."""
+    tr = TIME_RESOLUTIONS
+    res = tr[tr.index(max_period):tr.index(min_period) + 1]
+    return dtype_factory('timedelta64[{}]', res, TIME_RESOLUTIONS, endianness)
+
+
+@defines_dtype_strategy
+def string_dtypes(byte_strs=True, unicode_strs=True,
```

Comment:
I do still think these should be separate. It's more consistent with the rest of the API, and I don't really want to encourage bytes/unicode confusion.